hrjobcontract/CRM/Hrjobcontract/DAO/Base.php

```diff
@@ -299,12 +299,30 @@ public static function create($params)
                 $tableName . '_revision_id' => $params['jobcontract_revision_id'],
             ));
         }
-        
+
+        //setting the 'effective_date' if it's not set: Only executed for the
+        //HRJobDetails BAO.
+        if ($baoName == 'CRM_Hrjobcontract_BAO_HRJobDetails') {
```

Comment:
you can use the `$entityName` variable here and make this shorter.

The "Only executed for the HRJobDetails BAO" part is not really necessary, since that is clear from the `if`